streaming/src/main/scala/com/stratio/crossdata/streaming/helpers/CrossdataStreamingHelper.scala

```diff
@@ -46,14 +51,33 @@ object CrossdataStreamingHelper extends SparkLoggerComponent {
     val kafkaInput = new KafkaInput(kafkaOptions)
     val kafkaDStream = toWindowDStream(kafkaInput.createStream(streamingContext), ephemeralTable.options)
 
+    // DStream.foreachRDD is a method that is executed in the Spark Driver if and when an output action is not called
+    // over a RDD. Thus, the value countdowns can be used inside.
+    // More information here:
+    // http://spark.apache.org/docs/latest/streaming-programming-guide.html#design-patterns-for-using-foreachrdd
     kafkaDStream.foreachRDD { rdd =>
       if (rdd.take(1).length > 0) {
         val ephemeralQueries = CrossdataStatusHelper.queriesFromEphemeralTable(zookeeperConf, ephemeralTable.name)
 
         if (ephemeralQueries.nonEmpty) {
-          ephemeralQueries.foreach(ephemeralQuery =>
-            executeQuery(rdd, ephemeralQuery, ephemeralTable, kafkaOptions, zookeeperConf, crossdataCatalogConf)
-          )
+          ephemeralQueries.foreach( ephemeralQuery => {
+            val alias = ephemeralQuery.alias
+            if(!countdowns.contains(alias)){
```

Comment:
Can be written in a shorter form:

` countdowns(alias) = m.getOrElseUpdate(alias, ephemeralQuery.window / sparkStreamingWindow+1)-1`